bin/smack.py

```diff
@@ -386,7 +477,15 @@ def smackdOutput(corralOutput):
         colno = int(traceMatch.group(3))
         threadid = int(traceMatch.group(4))
         desc = str(traceMatch.group(6))
-        trace = { 'threadid': threadid, 'file': filename, 'line': lineno, 'column': colno, 'description': '' if desc == 'None' else desc }
+        assm = ''
```

Comment:
For each trace line that we parse, the witness file needs to know if there is an assumption made by the tool regarding a specific variable, allowing the witness checker to make the same assumption and thus prune the search space.

With smack, we don't see a variable assumption on each trace line.  Clearly, then for those without, we can't add a variable value assumption to the witness file.  For such trace lines, this change adds the assumption as an empty string (this makes it easier when reading the json later - we know the `'assumption'` child node will always exist, so we can just check to see if it has a value).  For trace lines that DO have a variable value assumption, the symbol and value are extracted from the `traceAssumeMatch` groups, and added as the `'assumption'` child to the json `'trace'` node.

I split the assignment of `trace` on to multiple lines because it was becoming unwieldy with the addition of the `'assumption'` child node.